src/pages/alerts.astro

```diff
@@ -41,17 +41,23 @@ import "../styles/alerts.css";
       alertsMainElem.prepend(alertElement);
     });
 
-  ws.onmessage = (m) => {
-    if (["ping", "welcome"].includes(m.data)) return;
+  ws.onmessage = (msg: MessageEvent<string>) => {
+    if (msg.data === "ping" || msg.data.startsWith("welcome")) return;
 
-    const alert = JSON.parse(m.data) as Alert;
+    const alert = JSON.parse(msg.data) as Alert<AlertType>;
 
     alertHistory.push(alert);
     localStorage.setItem("alertHistory", JSON.stringify(alertHistory));
 
     const alertElement = document.createElement("div");
-    alertElement.className = alert.data.status;
-    alertElement.innerHTML = alert.data.reason;
+
+    if (isAppealAlert(alert)) {
+      // using className like this will work for now, but with more alert types it'll become spaghetti code
+      alertElement.className = alert.data.status;
+      alertElement.innerHTML = alert.data.reason;
```

Comment:
You can use:

```ts
alertElement.classList.add(alert.data.status)
```

to achieve this. If you want the different alert types to be styled differently then you have to modify the element itself in some way for the CSS to pick it up and style it, whether that's using HTML datasets, or classes.